src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurRequestExtractFileServiceImpl.java

```diff
@@ -0,0 +1,172 @@
+        moveFile(pathAndFileName, getRejectedDirectoryName());
+    }
+
+    public void performAcceptedRequestExtractFileTasks(String pathAndFileName) {
+        moveFile(pathAndFileName, getAcceptedDirectoryName());
+    }
+
+    public void performDoneFileTasks(String requestExtractFileName) {
+        File doneFile = new File(StringUtils.substringBeforeLast(requestExtractFileName, ".") + ".done");
+        if (doneFile.exists()) {
+            doneFile.delete();
+        }
+    }
+
+    private void moveFile(String currentPathAndFile, String newPath) {
+        String newPathFileNameAndExtension = new String(newPath + ConcurConstants.FORWARD_SLASH + (StringUtils.substringAfterLast(currentPathAndFile, ConcurConstants.FORWARD_SLASH)));
+        LOG.info("Moving file From=" +currentPathAndFile+ "=   To=" +newPathFileNameAndExtension+ "=");
+        try {
+            FileUtils.moveFile(new File(currentPathAndFile), new File(newPathFileNameAndExtension));
+        } catch (IOException e) {
+            LOG.error("Unable to move file From=" + currentPathAndFile + "=  To=" + newPathFileNameAndExtension, e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void setBatchInputFileService(BatchInputFileService batchInputFileService) {
+        this.batchInputFileService = batchInputFileService;
+    }
+
+    public BatchInputFileService getBatchInputFileService() {
+        return batchInputFileService;
+    }
+
+    public void setBatchInputFileType(BatchInputFileType batchInputFileType) {
+        this.batchInputFileType = batchInputFileType;
+    }
+
+    public BatchInputFileType getBatchInputFileType() {
+        return batchInputFileType;
+    }
+
+    public void setAcceptedDirectoryName(String acceptedDirectoryName) {
+        this.acceptedDirectoryName = acceptedDirectoryName;
+    }
+
+    public String getAcceptedDirectoryName() {
+        return this.acceptedDirectoryName;
+    }
+
+    public void setRejectedDirectoryName(String rejectedDirectoryName) {
+        this.rejectedDirectoryName = rejectedDirectoryName;
+    }
+
+    public String getRejectedDirectoryName() {
+        return this.rejectedDirectoryName;
+    }
+
+    public void setConcurRequestExtractFileValidationService(ConcurRequestExtractFileValidationService concurRequestExtractFileValidationService) {
+        this.concurRequestExtractFileValidationService = concurRequestExtractFileValidationService;
+    }
+
+    public ConcurRequestExtractFileValidationService getConcurRequestExtractFileValidationService() {
+        return this.concurRequestExtractFileValidationService;
+    }
+
+    private Object loadFile(String fileName) {
+        byte[] fileByteContent = safelyLoadFileBytes(fileName);
+        Object parsedObject;
+        parsedObject = getBatchInputFileService().parse(getBatchInputFileType(), fileByteContent);
+        return parsedObject;
+    }
+
+    private byte[] safelyLoadFileBytes(String fileName) {
```

Comment:
I see both you and Jay are using this method to load file bytes and I have seen similar code in other batch jobs. Let's extract this method into a utility service so that it can be re-used. Thank you!